### PR TITLE
Implementing a dummy location manager

### PR DESCRIPTION
**Pull request by Adam-GitHub0**
Implementing a dummy location manager to connect the location to the database and add comments and documentation

### CODE REVIEW

File: app/src/main/java/com/github/versus/LocationFragment.java [hunk 1]
```diff
@@ -108,12 +115,11 @@ public class LocationFragment extends Fragment implements OnMapReadyCallback {
...
     private static ListView listView ;
     private static boolean hasLocations = false;
-    private static Circle mapCircle;
-    private List<CustomPlace> customPlaces = Arrays.asList(new CustomPlace("UNIL Football", "UNIL Football", new LatLng(46.519385, 6.580856)),
-            new CustomPlace("Chavannes Football", "Chavannes Football", new LatLng(46.52527373363714, 6.57366257779824)),
-            new CustomPlace("Bassenges Football", "Bassenges Football", new LatLng(46.52309381914529, 6.5608807098372175)),
-            new CustomPlace("GooglePlex Football", "Google Football", new LatLng(37.422083, -122.082555)));
+    private Marker lastClickedMarker;
+    private Circle lastDrawnCircle;
+    private DummyLocationManager dummyLocationManager;
```
Comment (hamzaremmal) [inline app/src/main/java/com/github/versus/LocationFragment.java]: Isn't it better to have an higher abstraction of this type (an interface perhaps) ?

File: app/src/main/java/com/github/versus/LocationFragment.java [hunk 2]
```diff
@@ -253,18 +268,81 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
+
+        // Add a new marker to the map at the clicked position
+        lastClickedMarker = map.addMarker(new MarkerOptions()
+                .position(clickedPosition)
+                .icon(BitmapDescriptorFactory.defaultMarker(BitmapDescriptorFactory.HUE_AZURE))
+                .title("Clicked Location"));
+
+        // Iterate through the custom places and find the ones within the threshold distance
+        List<CustomPlace> nearbyFields = new ArrayList<>();
+        for (CustomPlace customPlace : customPlaces) {
```
Comment (hamzaremmal) [inline app/src/main/java/com/github/versus/LocationFragment.java]: Maybe that's just my obsession with functional programming, but isn't it better to use [streams](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/stream/Stream.html) ?

The code will look like :

```java
var nearbyFields = customPlace.stream()
   .filter(place -> haversineDistance(clickedPosition, place.latLng) <= thresholdDistance)
   .toList()
```

File: app/src/main/java/com/github/versus/db/DummyLocationManager.java
Comment (hamzaremmal) [inline app/src/main/java/com/github/versus/db/DummyLocationManager.java:1]: Will this be replaced by a proper implementation later ?

Comment (Adam-GitHub0) [reply app/src/main/java/com/github/versus/db/DummyLocationManager.java:1]: Yes ! Based on this , Abdessalam will implement it in the database !

File: app/src/main/java/com/github/versus/LocationFragment.java (hunk 2 above)
Comment (Adam-GitHub0) [reply app/src/main/java/com/github/versus/LocationFragment.java]: Nice , it seems more concise this way , lets do this!

File: app/src/main/java/com/github/versus/LocationFragment.java (hunk 1 above)
Comment (Adam-GitHub0) [reply app/src/main/java/com/github/versus/LocationFragment.java]: Yes , I agree ! I've changed it in my last commit !

**Review by hamzaremmal — APPROVED** [review]

**Pull request merged**